mne/viz/evoked.py

```diff
@@ -1200,3 +1200,439 @@ def plot_evoked_joint(evoked, times="peaks", title='', picks=None,
+        for each channel equals the pyplot default.
+    invert_y : bool
+        If True, negative values are plotted up (as is sometimes done
+        for ERPs out of tradition). Defaults to False.
+    axes : None | `matplotlib.pyplot.axes` instance | list of `axes`
+        What axes to plot to. If None, a new axes is created.
+        When plotting multiple channel types, can also be a list of axes, one
+        per channel type.
+    title : None | str
+        If str, will be plotted as figure title. If None, the channel
+        names will be shown.
+    show : bool
+        If True, show the figure.
+
+    Returns
+    -------
+    fig : Figure | list of Figures
+        The figure(s) in which the plot is drawn.
+    """
+    import matplotlib.pyplot as plt
+    from ..evoked import Evoked, combine_evoked
+
+    # set up labels and instances
+    if isinstance(evokeds, Evoked):
+        evokeds = dict(Evoked=evokeds)  # title becomes 'Evoked'
+    elif not isinstance(evokeds, dict):
+        evokeds = dict((str(ii + 1), evoked)
+                       for ii, evoked in enumerate(evokeds))
+    conditions = sorted(list(evokeds.keys()))
+
+    # get and set a few limits and variables (times, channels, units)
+    example = (evokeds[conditions[0]]
+               if isinstance(evokeds[conditions[0]], Evoked)
+               else evokeds[conditions[0]][0])
+    if not isinstance(example, Evoked):
+        raise ValueError("evokeds must be an instance of mne.Evoked "
+                         "or a collection of mne.Evoked's")
+    times = example.times
+    tmin, tmax = times[0], times[-1]
+
+    if isinstance(picks, int):
+        picks = [picks]
+    elif len(picks) == 0:
+        warn("No picks, plotting the GFP ...")
+        gfp = True
+        picks = _pick_data_channels(example.info)
+
+    # deal with picks: infer indices and names
+    if gfp is True:
+        ch_names = ['Global Field Power']
+        if len(picks) < 2:
+            raise ValueError("A GFP with less than 2 channels doesn't work, "
+                             "please pick more channels.")
+    else:
+        if not isinstance(picks[0], int):
+            msg = "'picks' must be int or a list of int, not {0}."
+            raise ValueError(msg.format(type(picks)))
+        ch_names = [example.ch_names[pick] for pick in picks]
+    ch_types = list(set(channel_type(example.info, pick_)
+                    for pick_ in picks))
+    # XXX: could possibly be refactored; plot_joint is doing a similar thing
+    if any([type_ not in _DATA_CH_TYPES_SPLIT for type_ in ch_types]):
+        raise ValueError("Non-data channel picked.")
+    if len(ch_types) > 1:
+        warn("Multiple channel types selected, returning one figure per type.")
+        if axes is not None and len(axes) != len(ch_types):
+            msg = "Please provide one axis per channel type ({0} required)."
+            raise ValueError(msg.format(len(ch_types)))
```

Comment:
You could also do:
```python
from .utils import _validate_if_list_of_axes
_validate_if_list_of_axes(axes, obligatory_len=len(ch_types))
```
It does all the checks you do here and checks if each element of the list is an axes object too.